### PR TITLE
Logging adjustments

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes issue #144 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] I have added my relevant user information to `AUTHORS.md`

* **What kind of change does this PR introduce?:** 

This reimplements logging in CLISOPS using [Loguru](https://github.com/Delgan/loguru), an alternative logging library that is virtually identical by way of API but much more streamlined in its configuration.

The current logging setup initializes `Handlers` on import, which are then propagated to upstream libraries. This causes issues upstream when outputs to `stdout` and `stderr` can falsely cause build failures on CI and ReadTheDocs in other libraries and applications (see note about creating Handlers in libraries: https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library).

The changes presented here ~disable logging~ remove the rootlogger for CLISOPS by default, allowing for logging to be explicitly set for the library using a function (`clisops.utils.common.enable_logging()`), or by setting a handler in the `main()` loop of a running process using the `loguru`'s `logger` class (e.g. `from loguru import logger; logger.add(sink=sys.stdout)`).

* **Does this PR introduce a breaking change?:** 

Yes. Logging is no longer dependent on the standard `logging` library. In order to enable logging for CLISOPS, the called process must enable logging with `clisops.utils.common.enable_logging()` (This can be moved somewhere/renamed if needed). This will create the handlers for CLISOPS to output to `stdout`, stderr` (and elsewhere, if we want).

Since there are pytests relying on `caplog` and `loguru` does not write to the standard logging handler, I used a [pytest plugin](https://pypi.org/project/pytest-loguru/) that passes messages from loguru to `caplog`. For tests that examine the logger, there is no difference beyond ensuring that logging is enabled for clisops (i.e. `logger.enable("clisops")`)

* **Other information:** 
Relevant links:
 - https://github.com/Delgan/loguru/issues/349
 - https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library
 - https://github.com/mcarans/pytest-loguru

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -41,7 +41,7 @@ test = pytest
 
 [tool:pytest]
 collect_ignore = ["setup.py"]
-addopts = --verbose tests
+addopts = --verbose
```
Comment (Zeitsperre) [inline setup.cfg:44]: This setup previously prevented developers from running tests selectively. Now, tests can be run individually or by module.

**Review by huard — COMMENTED** [review]
No objection on my end, but it should be mentionned that as far as I know, we have no operational experience with loguru. Maybe it would be best to try it out on our end and iron out the kinks before putting this in prod. 

Also, I believe logging practices should be broadly similar across birdhouse libraries, so @cehbrecht would have a veto on this kind of change.

File: setup.cfg (hunk 1 above)
Comment (cehbrecht) [reply setup.cfg:44]: The other workaround is to use `pytest -c doesnotexist tests/test_only_me.py`

**Comment by cehbrecht** [comment]
If loguru makes are logging easier it would be nice. It has also something to handle in multiprocessing which we might want to use in pywps. 

I'm a bit worried about the transition of the codes and the activation of the loguru logging.

**Comment by Zeitsperre** [comment]
I wasn't going to plunge into the depths of asynchronous logging, but yes it looks as though there is robust support for safely-threaded logging in loguru. I'm leaving this here as an example, and perhaps we can talk about this change in our next meeting.

Logging in general is something that I would like us to make more use of on the Ouranos side. @huard mentioned that we could try deploying this on some Ouranos projects first to get a sense of its behaviour. I'll find an afternoon to give that a try.

**Comment by agstephens** [comment]
@Zeitsperre, this looks great. Let me know when it is ready to merge.

**Comment by Zeitsperre** [comment]
Hey Ag, there are a few final changes to this that I want to make, based on the work done in https://github.com/Ouranosinc/xclim/pull/1041. The biggest breaking change is that the default behaviour for logging should be set to 'disabled'; The running process/user should be responsible for enabling logging:
```
import sys
from loguru import logger

logger.enable("clisops")
logger.add(sys.stdout, level="INFO||DEBUG||etc.")
```
This whole process should also be documented for clarity as well, so I'll add something similar based on the other PR too. It should be finished today!

**Comment by Zeitsperre** [comment]
@agstephens This PR is now fully functioning for me!

**Comment by agstephens** [comment]
Hi @Zeitsperre , could you please fix the conflicts and then we can merge this one as well.

**Pull request merged**